src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from 'react';
+
+import { Container } from '@mui/system';
+import { Await, useLoaderData } from 'react-router-dom';
+import { style } from 'typestyle';
+
+import { useAppSelector } from '../../../hooks/app';
+import { Colors, Spacing } from '../../../styles';
+import type { SubtestsRevisionsHeader } from '../../../types/state';
+import DownloadButton from '.././DownloadButton';
+import SearchInput from '.././SearchInput';
+import { LoaderReturnValue } from '../subtestsLoader';
+import { LoaderReturnValue as OvertimeLoaderReturnValue } from '../subtestsOverTimeLoader';
+import SubtestsResultsTable from './SubtestsResultsTable';
+import SubtestsRevisionHeader from './SubtestsRevisionHeader';
+
+function SubtestsResultsMain() {
+  const { results } = useLoaderData() as LoaderReturnValue | OvertimeLoaderReturnValue;
+
+  const themeMode = useAppSelector((state) => state.theme.mode);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const subtestsHeader: SubtestsRevisionsHeader = {
+    suite: results[0].suite,
+    framework_id: results[0].framework_id,
+    test: results[0].test,
+    option_name: results[0].option_name,
+    extra_options: results[0].extra_options,
+    new_rev: results[0].new_rev,
+    new_repo: results[0].new_repository_name,
+    platform: results[0].platform,
+  };
+
+  const themeColor100 =
+    themeMode === 'light' ? Colors.Background300 : Colors.Background100Dark;
+
+  const styles = {
+    container: style({
+      backgroundColor: themeColor100,
+      margin: '0 auto',
+      marginBottom: '80px',
+    }),
+    title: style({
+      margin: 0,
+      marginBottom: Spacing.Medium,
+    }),
+    content: style({
+      display: 'flex',
+      justifyContent: 'flex-end',
+      alignItems: 'center',
+    }),
+  };
+
+  return (
+    <Container className={styles.container} data-testid='subtests-main'>
+      <Await resolve={results}>
```

Comment:
Removing Await will results in `TypeError: results is undefined`

Reply:
That's true, but that's not the right fix.
The error happens because you're using `useAsyncValue` in `SubtestsResultsTable`. `useAsyncValue` works with `Await`.
But `results` is not a promise! so there's no need to use `Await` and there's no need to use `useAsyncvalue`, you could use `useLoaderData` in `SubtestsResultsTable` or directly pass down `results` using a prop.